tutorials/mission_specific_analyses/rxte/visualize-rxte-spectra.md

```diff
@@ -0,0 +1,379 @@
+
+We've already figured out which HEASARC table to pull RXTE observation information from, and then used that table
+to identify the specific observations that might be relevant to our target source (Eta Car) - the next step is to
+pinpoint the exact files from each observation that we can use to visualize the spectral emission of our source.
+
+#### Setting up file search criteria
+
+:::{danger}
+This solution isn't necessarily what we want to do for the SciServer version, but I am still trying to figure out
+how exactly we have different solutions shown for different platforms.
+:::
+
+Just as in the last two steps, we're going to make use of PyVO. The difference is, rather than dealing with tables of
+observations, we now need to construct 'datalinks' to places where specific files for each observation are stored. In 
+this demonstration we're going to pull data from the HEASARC 'S3 bucket', an Amazon-hosted open-source dataset 
+containing all of HEASARC's data holdings. 
+
+As this demonstration is only concerned with visualizing RXTE PCA spectra, and there are both other instruments and 
+non-spectral data files associated with RXTE, we will use PyVO's semantics to filter out irrelevant files. The 
+`semantic_base` variable defines the **RXTE-PCA-specific** structure of the 'semantics' column contained in 
+all PyVO datalink tables, and the `target_prod` variable defines the semantics associated with the three files
+we're interested in for each observation:
+
+- The spectrum automatically generated for the target of the RXTE observation (*pha_s2*).
+- A background spectrum generated as a companion for the source spectrum (*pha_b2*).
+- The supporting file that defines the response curve (sensitivity over energy range) and redistribution matrix (a mapping of channel to energy) for the RXTE-PCA instrument during this observation (*rsp*).
+
+```{code-cell}
+semantic_base = "https://heasarc.gsfc.nasa.gov/xamin/jsp/products.jsp#xte.prod.pca."
+target_prod = ["pha_s2", "pha_b2", "rsp"]
+
+search_sem = [semantic_base + t_prod for t_prod in target_prod]
+search_sem
+```
+
+For our convenience, we also create a dictionary that will be used to map file descriptions taken from PyVO datalink
+tables to more Pythonic (and easier to type!) strings:
+
+```{code-cell}
+conv_desc = {'PCA Response Matrix': 'resp', 
+             'PCA Background Standard2 Spectra': 'bck_spec', 
+             'PCA Source Standard2 Spectra': 'src_spec'}
+```
+
+#### Running the search for datalinks to files of interest
+
+Following the setup we performed in the previous step, we can now search for the relevant datalinks. 
+
+<span style="color:red">This solution is currently too slow to include in the final version of this 
+demonstration, which I hope is due to my ignorance of PyVO</span>
+
+
+```{code-cell}
+rel_dls = {}
+
+with tqdm(desc="Identifying XTE PCA spectra S3 URIs", total=len(results)) as uri_prog:
```

Comment:
in astroquery
```python
links = Heasarc.locate_data(results)
```

But the datalinks service is slow for a large number of datalinks. Changes are needed in the backend.